src/main/java/com/openlattice/chronicle/controllers/legacy/ChronicleController.java

```diff
@@ -0,0 +1,66 @@
+package com.openlattice.chronicle.controllers.legacy;
+
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.SetMultimap;
+import com.openlattice.chronicle.ChronicleApi;
+import com.openlattice.chronicle.services.edm.EdmCacheManager;
+import com.openlattice.chronicle.services.enrollment.EnrollmentManager;
+import com.openlattice.chronicle.services.upload.AppDataUploadManager;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+@RestController
+@RequestMapping( ChronicleApi.CONTROLLER )
+public class ChronicleController implements ChronicleApi {
+
+    @Inject
+    private AppDataUploadManager dataUploadManager;
+
+    @Inject
+    private EdmCacheManager edmCacheManager;
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH,
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Integer upload(
+            @PathVariable( STUDY_ID ) UUID studyId,
+            @PathVariable( PARTICIPANT_ID ) String participantId,
+            @PathVariable( DATASOURCE_ID ) String datasourceId,
+            @RequestBody List<SetMultimap<UUID, Object>> data ) {
+
+        return dataUploadManager.upload( null, studyId, participantId, datasourceId, data );
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = EDM_PATH,
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Map<String, UUID> getPropertyTypeIds( @RequestBody Set<String> propertyTypeFqns ) {
```

Comment:
This should probably take a `Set<FullQualifiedName>` as an input param. Jackson should be able to deserialize valid FQNs coming in from requests.

Reply:
I totally agree. However, this endpoint is consumed by the android app...therefore changing this would need to be accompanied by changes to the app... which is problematic because we have no control over users updating the app.

Reply:
As long as the apps are passsing FQN strings then it doesn't matter how the app represents it internally -- all requests are deserialized to the expected format using jackson, so as long as the incoming values are of the form "blah.blah" then if you set the type as FullQualifiedName then jackson will read those values as that type